src/components/position_calc/NLS_position_calc.py

```diff
@@ -0,0 +1,53 @@
+from components.position_calc import position_calc_utils
+import numpy as np
+from simulator_main import sim_config as cfg
+from sim_utils.common_types import *
+
+class NLS_position_calc:
+    
+    def __init__(self, name = "NLS_position_calc", optimization_type = OptimizationType.nelder_mead, 
+                initial_guess = CylindricalPosition(0,0,0), is_polar=True):
+        self.name = name
+        self.optimization_type = optimization_type
+        if (is_polar):
```

Comment:
If you switch to only a dictionary as input parameter, you could do something like:

is_polar = parameter_dict.get("is_polar", True) which gives you the correct behaviour if you want is_polar to be optional

(same with other optional parameters)